WireAPI/Sources/WireAPI/APIs/ConnectionsAPI/ConnectionsAPIV0.swift

```diff
@@ -0,0 +1,130 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class ConnectionsAPIV0: ConnectionsAPI, VersionedAPI {
+
+    enum Constants {
+        static let maxConnectionsCount = 500
+    }
+
+    let httpClient: HTTPClient
+    let fetchLimit: Int
+    let decoder = ResponsePayloadDecoder(decoder: .defaultDecoder)
+
+    convenience init(httpClient: HTTPClient) {
+        self.init(httpClient: httpClient, fetchLimit: Constants.maxConnectionsCount)
+    }
+
+    init(httpClient: HTTPClient, fetchLimit: Int) {
+        self.httpClient = httpClient
+        self.fetchLimit = Constants.maxConnectionsCount
+    }
+
+    var apiVersion: APIVersion {
+        .v0
+    }
+
+    var resourcePath: String {
+        "\(pathPrefix)/list-connections/"
+    }
+
+    func fetchConnections() async throws -> PayloadPager<Connection> {
+
+        let pager = PayloadPager<Connection> { start in
+
+            // body Params
+            let params = PaginationRequest(pagingState: start, size: Constants.maxConnectionsCount)
+            let body = try JSONEncoder.defaultEncoder.encode(params)
+
+            let request = HTTPRequest(
+                path: self.resourcePath,
+                method: .post,
+                body: body
+            )
+            let response = try await self.httpClient.executeRequest(request)
+
+            // Parse response
+            let responsePayload = try ResponseParser()
+                .success(code: 200, type: PaginatedConnectionList.self)
+                .failure(code: 400, error: ConnectionsAPIError.invalidBody)
+                .parse(response)
+
+            return PayloadPager<Connection>.Page(
+                element: responsePayload.connections.map { $0.toAPIModel() },
+                hasMore: responsePayload.hasMore,
+                nextStart: responsePayload.pagingState
+            )
+        }
+
+        return pager
+    }
+}
+
+private struct PaginatedConnectionList: Decodable, ToAPIModelConvertible {
```

Comment:
```suggestion
private struct PaginatedConnectionListV0: Decodable, ToAPIModelConvertible {
```

Reply:
Or should this be a public struct and not bound the V0?

Reply:
No it can stay private. I had a small try to have some generic struct that would look like this:
```
    struct PaginatedList<T: Decodable>: Codable {

        enum CodingKeys: String, CodingKey {
            case connections // need to replace this
            case pagingState = "paging_state"
            case hasMore = "has_more"
        }


        let content: [T]
        let pagingState: String
        let hasMore: Bool
    }
```
I don't know how many endpoints use this paging has_more. 
A solution would be to implement decodable and use one parameter to define the key (connections)